### ISSUE TITLE
Enum in Trait Client Gen Failure

### ISSUE THREAD

**Issue by gurumeditator** [atomicbits/scraml#116]
Hi, I am receiving an error when generating a client (0.5.2) given the following trait fragment:

```
queryParameters:

  filter_op:
    description: |

      *The match operation (operator)*

    example: filter_op=GREATER_THAN
    enum: [GREATER_THAN, GREATER_THAN_EQUAL]
...
```

Error received:
```
[ERROR] Failed to execute goal io.atomicbits:scraml-maven-plugin:0.5.2:scraml (generate-client-sources) on project MMISprint19: Execution generate-client-sources of goal io.atomicbits:scraml-maven-plugin:0.5.2:scraml failed: EnumType(ImplicitId,List(GREATER_THAN, GREATER_THAN_EQUAL),None,RamlModel) (of class io.atomicbits.scraml.ramlparser.model.types.EnumType) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.atomicbits:scraml-maven-plugin:0.5.2:scraml (generate-client-sources) on project MMISprint19: Execution generate-client-sources of goal io.atomicbits:scraml-maven-plugin:0.5.2:scraml failed: EnumType(ImplicitId,List(GREATER_THAN, GREATER_THAN_EQUAL),None,RamlModel) (of class io.atomicbits.scraml.ramlparser.model.types.EnumType)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution generate-client-sources of goal io.atomicbits:scraml-maven-plugin:0.5.2:scraml failed: EnumType(ImplicitId,List(GREATER_THAN, GREATER_THAN_EQUAL),None,RamlModel) (of class io.atomicbits.scraml.ramlparser.model.types.EnumType)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        ... 20 more
Caused by: scala.MatchError: EnumType(ImplicitId,List(GREATER_THAN, GREATER_THAN_EQUAL),None,RamlModel) (of class io.atomicbits.scraml.ramlparser.model.types.EnumType)
        at io.atomicbits.scraml.generator.codegen.JavaActionCode$.expandQueryOrFormParameterAsMethodParameter(JavaActionCode.scala:164)
        at io.atomicbits.scraml.generator.codegen.ActionFunctionGenerator$$anonfun$3.apply(ActionFunctionGenerator.scala:104)
        at io.atomicbits.scraml.generator.codegen.ActionFunctionGenerator$$anonfun$3.apply(ActionFunctionGenerator.scala:104)
        at scala.collection.immutable.List.map(List.scala:277)
        at io.atomicbits.scraml.generator.codegen.ActionFunctionGenerator.generateBodyAction(ActionFunctionGenerator.scala:104)
        at io.atomicbits.scraml.generator.codegen.ActionFunctionGenerator.generate(ActionFunctionGenerator.scala:38)
        at io.atomicbits.scraml.generator.codegen.ActionGenerator$$anonfun$11.apply(ActionGenerator.scala:144)
        at io.atomicbits.scraml.generator.codegen.ActionGenerator$$anonfun$11.apply(ActionGenerator.scala:144)
        at scala.collection.immutable.List.flatMap(List.scala:327)
        at io.atomicbits.scraml.generator.codegen.ActionGenerator.io$atomicbits$scraml$generator$codegen$ActionGenerator$$expandAcceptHeaderMap(ActionGenerator.scala:144)
        at io.atomicbits.scraml.generator.codegen.ActionGenerator$$anonfun$9.apply(ActionGenerator.scala:98)
        at io.atomicbits.scraml.generator.codegen.ActionGenerator$$anonfun$9.apply(ActionGenerator.scala:96)
        at scala.collection.immutable.List.map(List.scala:273)
        at io.atomicbits.scraml.generator.codegen.ActionGenerator.generateActionFunctions(ActionGenerator.scala:96)
        at io.atomicbits.scraml.generator.codegen.JavaResourceClassGenerator$.io$atomicbits$scraml$generator$codegen$JavaResourceClassGenerator$$generateResourceClassesHelper$1(JavaResourceClassGenerator.scala:124)
        at io.atomicbits.scraml.generator.codegen.JavaResourceClassGenerator$$anonfun$4.apply(JavaResourceClassGenerator.scala:175)
        at io.atomicbits.scraml.generator.codegen.JavaResourceClassGenerator$$anonfun$4.apply(JavaResourceClassGenerator.scala:175)
        at scala.collection.immutable.List.flatMap(List.scala:327)
        at io.atomicbits.scraml.generator.codegen.JavaResourceClassGenerator$.io$atomicbits$scraml$generator$codegen$JavaResourceClassGenerator$$generateResourceClassesHelper$1(JavaResourceClassGenerator.scala:175)
        at io.atomicbits.scraml.generator.codegen.JavaResourceClassGenerator$$anonfun$generateResourceClasses$1.apply(JavaResourceClassGenerator.scala:256)
        at io.atomicbits.scraml.generator.codegen.JavaResourceClassGenerator$$anonfun$generateResourceClasses$1.apply(JavaResourceClassGenerator.scala:256)
        at scala.collection.immutable.List.flatMap(List.scala:327)
        at io.atomicbits.scraml.generator.codegen.JavaResourceClassGenerator$.generateResourceClasses(JavaResourceClassGenerator.scala:256)
        at io.atomicbits.scraml.generator.ScramlGenerator$.generateClassReps(ScramlGenerator.scala:155)
        at io.atomicbits.scraml.generator.ScramlGenerator$.generateFor(ScramlGenerator.scala:89)
        at io.atomicbits.scraml.generator.ScramlGenerator$.generateJavaCode(ScramlGenerator.scala:64)
        at io.atomicbits.scraml.generator.ScramlGenerator.generateJavaCode(ScramlGenerator.scala)
        at io.atomicbits.scraml.mvnplugin.ScramlMojo.execute(ScramlMojo.java:110)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        ... 21 more
```

**Comment by rigolepe** [comment]
Looks like a bug indeed, I will have a look at it asap.

**Comment by rigolepe** [comment]
Apologies for the delay here, there was a big refactoring that had to come first. The bug is fixed and will be released in 0.7.
